medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -247,6 +278,193 @@ mod test {
+        let storage_client = given_a_storage_client_that_cannot_delete_blob();
+        given_the_necessary_env_vars_are_initialised();
+
+        let result = block_on(process_delete_message(
+            removeable_message,
+            storage_client,
+            search_client,
+        ));
+        unset_process_delete_message_env_vars();
+
+        match result {
+            Ok(_) => panic!("Error expected"),
+            Err(e) => {
+                assert_eq!(
+                    e.to_string(),
+                    ProcessMessageError::FailedDeletingBlob(
+                        "storage_name".to_string(),
+                        "Generic error: blob could not be deleted".to_string()
+                    )
+                    .to_string()
+                );
+            }
+        }
+    }
+
+    #[test]
+    fn failure_to_restore_index_returns_expected_error() {
+        let removeable_message = given_we_have_a_delete_message().message;
+        let search_client = given_a_search_client_that_cannot_restore_index();
+        let storage_client = given_a_storage_client_that_cannot_delete_blob();
+        given_the_necessary_env_vars_are_initialised();
+
+        let result = block_on(process_delete_message(
+            removeable_message,
+            storage_client,
+            search_client,
+        ));
+        unset_process_delete_message_env_vars();
+
+        match result {
+            Ok(_) => panic!("Error expected"),
+            Err(e) => {
+                assert_eq!(
+                    e.to_string(),
+                    ProcessMessageError::FailedRestoringIndex(
+                        "storage_name".to_string(),
+                        "Index could not be created".to_string()
+                    )
+                    .to_string()
+                );
+            }
+        }
+    }
+
+    #[test]
+    fn failure_to_delete_index_returns_expected_error() {
+        let removeable_message = given_we_have_a_delete_message().message;
+        let search_client = given_a_search_client_that_cannot_delete_index();
+        let storage_client = given_a_storage_client();
+        given_the_necessary_env_vars_are_initialised();
+
+        let result = block_on(process_delete_message(
+            removeable_message,
+            storage_client,
+            search_client,
+        ));
+        unset_process_delete_message_env_vars();
+
+        match result {
+            Ok(_) => panic!("Error expected"),
+            Err(e) => {
+                assert_eq!(
+                    e.to_string(),
+                    ProcessMessageError::Generic(anyhow!("Index could not be deleted")).to_string()
+                );
+            }
+        }
+    }
+
+    fn given_the_necessary_env_vars_are_initialised() {
+        env::set_var("STORAGE_CONTAINER", "storage_container");
+    }
+
+    fn unset_process_delete_message_env_vars() {
+        env::set_var("STORAGE_CONTAINER", "");
```

Comment:
This doesn't do what it says. If we want to unset, can we use [`remove_var`](https://doc.rust-lang.org/std/env/fn.remove_var.html)?

Reply:
Set to use `remove_var`

Reply:
I had to remove this unsetting in the end as it seems like tests are multi-threaded so this currently sometimes unsets it for other tests that rely on it. We'll have to come up with a better way of making it thread safe, perhaps with test suite-level set-up and tear-down. I'll create a ticket to address this.